src/evict.c

```diff
@@ -389,7 +389,7 @@ size_t freeMemoryGetNotCountedMemory(void) {
  *              limit.
  *              (Populated both for C_ERR and C_OK)
  */
-int getMaxmemoryState(size_t *total, size_t *logical, size_t *tofree, float *level) {
+int getMaxmemoryState(size_t *total, size_t *logical, size_t *tofree, float *level, unsigned long long checked_maxmemory) {
```

Comment:
Would `maxmemorry` work for you? I feel that `checked_memory` introduces yet another concept ...
```suggestion
int getMaxmemoryState(size_t *total, size_t *logical, size_t *tofree, float *level, unsigned long long maxmemory) {
```

Reply:
it makes sense to me. but i need to update the comment before if (mem_used <= maxmemory) return C_OK; as following:

 /* if function parameter 'maxmemory' is equal to maxmemory and mem_used > maxmemory then OOM */
    /* if function parameter 'maxmemory' is equal to maxmemory_soft and mem_used > function parameter 'maxmemory' then there is no OOM but eviction happens */